File: 1686-stone-game-vi/1686-stone-game-vi.cpp
class Solution {
public:
    int stoneGameVI(vector<int>& A, vector<int>& B) {
        vector<vector<int>> C;
        int res[2] = {0, 0}, n = A.size();
        for (int i = 0; i < n; ++i)
            C.push_back({ -A[i] - B[i], A[i], B[i]});
        
        sort(begin(C), end(C));
        for (int i = 0; i < n; ++i)
            res[i % 2] += C[i][1 + i % 2];
        return res[0] == res[1] ? 0 : res[0] > res[1] ? 1 : -1;
    }
};

File: 1686-stone-game-vi/1686-stone-game-vi.java
class Solution {
    public int stoneGameVI(int[] A, int[] B) {
        int n = A.length;
        int[][] sums = new int[n][];
        for (int i = 0; i < n; i++) {
            sums[i] = new int[] {A[i] + B[i], A[i], B[i]};
        }
        Arrays.sort(sums, (a, b) -> Integer.compare(b[0], a[0]));
        int a = 0;
        int b = 0;
        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                a += sums[i][1];
            } else {
                b += sums[i][2];
            }
        }
        return Integer.compare(a, b);
    }
}